2020-01-Winter/4_trees/Kth_Smallest_Element_in_BST/recursion_KthSmallest_in_BST.py

```diff
@@ -0,0 +1,43 @@
+class TreeNode:
+     def __init__(self, x):
+         self.val = x
+         self.left = None
+         self.right = None
+
+def findNode(node, count, result):
+  if result[0] != -1:
+      return
+  if node.left:
+      findNode(node.left,count,result)
+  count[0] -= 1
+  if count[0] == 0:
+      result[0] = node.val
+      return
+  if node.right:
+      findNode(node.right,count, result)
+
+
+def kthSmallest(root: TreeNode, k: int) -> int:
+  count = [k]
+  #why use list when it's only 1 element? because list is mutable in python.
+  #So we can "pass by reference", while an int or float is immutable and cannot pass by reference.
+  result = [-1]
```

Comment:
Match source code file to README version.
```suggestion
  result = []
```